src/providers/base_java.js

```diff
@@ -112,16 +107,17 @@ export default class Base_Java {
 	}
 
 	/** this method invokes command string in a process in a synchronous way.
-	 * @param cmdString - the command to be invoked
-	 * @param errorMessage - the message to print to an exception thrown to client in case of error
+	 * @param bin - the command to be invoked
+	 * @param args - the args to pass to the binary
+	 * @param callback - function to invoke if an error was thrown
 	 * @protected
 	 */
-	_invokeCommand(cmdString, errorMessage) {
-		execSync(cmdString, err => {
-			if (err) {
-				throw new Error(`${errorMessage}`)
-			}
-		})
+	_invokeCommand(bin, args, callback) {
+		try {
+			execFileSync(bin, args)
```

Comment:
@ruromero I think we might want to pass `{ stdio: 'pipe' }` here, from some testing it seems like stderr is piped directly to the parent process' stderr (contrary to what the docs say :thinking:)

Reply:
I see indeed it should be the default. Can be that it depends on the node version?

Reply:
I've tested on node 22 and 18, as well as dnf provided vs nix provided. for both the behaviour appears to be `[null, 'pipe', 'inherit']`. Very odd. Ill enforce it to `pipe` to be sure

Reply:
Fine by me